modules/040-node-manager/hooks/handle_draining.go

```diff
@@ -107,9 +123,19 @@ func handleDraining(input *go_hook.HookInput, dc dependency.Container) error {
 	snap := input.Snapshots["nodes_for_draining"]
 	for _, s := range snap {
 		dNode := s.(drainingNode)
-		if !dNode.IsDraining {
+		if !dNode.isDraining() {
+			// if node has drained annotation (`user` source) and .spec.unscheduled == false - remove the annotation
+			if !dNode.Unschedulable && dNode.DrainedSource == "user" {
```

Comment:
What about `bashible` source or any other source? Where are they handled?

Reply:
In the update_approval hook